connectors/src/connectors/slack/temporal/activities.ts

```diff
@@ -119,6 +127,48 @@ interface SyncChannelRes {
   weeksSynced: Record<number, boolean>;
 }
 
+export async function joinChannel(
+  connectorId: ModelId,
+  channelId: string
+): Promise<"ok" | "already_joined" | "cant_join"> {
```

Comment:
This is a bit of a shortcut here. True it's convenient but it's not exactly standard. I think this should return channel information or a typed error of which it is easy to extract the reason of not joining?

Reply:
I will re-structure it to a lib + activity. It was initially only an activity.